Modules/_uuidmodule.c

```diff
@@ -91,45 +121,1730 @@ py_windows_has_stable_node(void)
+    EXTRACT_FIELD(
+        1, (1ULL << 16) - 1, "field 2 out of range (need a 16-bit value)",
+        uint16_t, time_mid
+    );
+    EXTRACT_FIELD(
+        2, (1ULL << 16) - 1, "field 3 out of range (need a 16-bit value)",
+        uint16_t, time_hi_version
+    );
+    EXTRACT_FIELD(
+        3, (1ULL << 8) - 1, "field 4 out of range (need an 8-bit value)",
+        uint8_t, clock_seq_hi_variant
+    );
+    EXTRACT_FIELD(
+        4, (1ULL << 8) - 1, "field 5 out of range (need an 8-bit value)",
+        uint8_t, clock_seq_low
+    );
+    EXTRACT_FIELD(
+        5, (1ULL << 48) - 1, "field 6 out of range (need a 48-bit value)",
+        uint64_t, node
+    );
+
+    self->bytes[0] = time_low >> 24;
+    self->bytes[1] = time_low >> 16;
+    self->bytes[2] = time_low >> 8;
+    self->bytes[3] = time_low;
+
+    self->bytes[4] = time_mid >> 8;
+    self->bytes[5] = time_mid;
+
+    self->bytes[6] = time_hi_version >> 8;
+    self->bytes[7] = time_hi_version;
+
+    self->bytes[8] = clock_seq_hi_variant;
+
+    self->bytes[9] = clock_seq_low;
+
+    self->bytes[10] = node >> 40;
+    self->bytes[11] = node >> 32;
+    self->bytes[12] = node >> 24;
+    self->bytes[13] = node >> 16;
+    self->bytes[14] = node >> 8;
+    self->bytes[15] = node;
+
+    return 0;
+}
+
+static PyObject *
+get_int(uuidobject *self)
+{
+    return _PyLong_FromByteArray((unsigned char *)self->bytes, 16, 0, 0);
+}
+
+static PyObject *
+get_uuidmod_attr(uuid_state *state, const char *name)
+{
+    PyObject *mod = PyImport_ImportModule("uuid");
+    if (mod == NULL) {
+        return NULL;
+    }
+
+    PyObject *ret = PyObject_GetAttrString(mod, name);
+    Py_DECREF(mod);
+    return ret;
+}
+
+static PyObject *
+get_SafeUUID(uuid_state *state)
+{
+    if (state->safe_uuid != NULL) {
+        return Py_NewRef(state->safe_uuid);
+    }
+
+    state->safe_uuid = get_uuidmod_attr(state, "SafeUUID");
+    if (state->safe_uuid == NULL) {
+        return NULL;
+    }
+    return Py_NewRef(state->safe_uuid);
+}
+
+static uuidobject *
+make_uuid(PyTypeObject *type)
+{
+    uuidobject *self = NULL;
+    uuid_state *state = get_uuid_state_by_cls(type);
+
+    Py_BEGIN_CRITICAL_SECTION(type);
+    if (state->freelist_size > 0) {
+        self = state->freelist;
+        state->freelist = (uuidobject *)self->weakreflist;
+        state->freelist_size--;
+    }
+    Py_END_CRITICAL_SECTION();
+
+    if (self != NULL) {
+        // Reinitialize the object from freelist
+        _Py_NewReference((PyObject *)self);
+    }
+    else {
+        self = PyObject_New(uuidobject, type);
+        if (self == NULL) {
+            return NULL;
+        }
+    }
+
+    self->is_safe = NULL;
+
+    self->weakreflist = NULL;
+    self->cached_hash = -1;
+
+    return self;
+}
+
+static PyObject *
+Uuid_alloc(PyTypeObject *type, PyObject *args, PyObject *kwargs)
+{
+    uuidobject *self = make_uuid(type);
+    if (self == NULL) {
+        return NULL;
+    }
+    memset(self->bytes, 0, 16);
```

Comment:
nipitck: would it be possible to add a UUID_SIZE constant (`#define UUID_SIZE 16`) and use it instead of hardcoded number 16?

```suggestion
    memset(self->bytes, 0, UUID_SIZE);
```